ecs-agent/netlib/platform/common_linux.go

```diff
@@ -205,3 +259,352 @@ func (c *common) buildSingleNSNetConfig(
+// inside a network namespace.
+func (c *common) configureInterface(
+	ctx context.Context,
+	netNSPath string,
+	iface *networkinterface.NetworkInterface,
+) error {
+	var err error
+	switch iface.InterfaceAssociationProtocol {
+	case networkinterface.DefaultInterfaceAssociationProtocol:
+		err = c.configureRegularENI(ctx, netNSPath, iface)
+	case networkinterface.VLANInterfaceAssociationProtocol:
+		err = c.configureBranchENI(ctx, netNSPath, iface)
+	default:
+		err = errors.New("invalid interface association protocol %s" + iface.InterfaceAssociationProtocol)
+	}
+	return err
+}
+
+// configureRegularENI configures a network interface for an ENI.
+func (c *common) configureRegularENI(ctx context.Context, netNSPath string, eni *networkinterface.NetworkInterface) error {
+	var cniNetConf []ecscni.PluginConfig
+	var add bool
+	var err error
+
+	c.os.Setenv(CNIPluginLogFileEnv, ecscni.PluginLogPath)
+	c.os.Setenv(IPAMDataPathEnv, filepath.Join(c.stateDBDir, IPAMDataFileName))
+
+	switch eni.DesiredStatus {
+	case status.NetworkReadyPull:
+		// The task metadata interface setup by bridge plugin is required only for the primary ENI.
+		if eni.IsPrimary() {
+			cniNetConf = append(cniNetConf, createBridgePluginConfig(netNSPath))
+		}
+		cniNetConf = append(cniNetConf, createENIPluginConfigs(netNSPath, eni))
+		add = true
+	case status.NetworkDeleted:
+		// Regular ENIs are used in single-use warmpool instances, so cleanup isn't necessary.
+		cniNetConf = nil
+		add = false
+	}
+
+	_, err = c.executeCNIPlugin(ctx, add, cniNetConf...)
+	if err != nil {
+		err = errors.Wrap(err, "failed to setup regular eni")
+	}
+
+	return err
+}
+
+// configureBranchENI configures a network interface for a branch ENI.
+func (c *common) configureBranchENI(ctx context.Context, netNSPath string, eni *networkinterface.NetworkInterface) error {
+	var cniNetConf ecscni.PluginConfig
+	var err error
+	add := true
+
+	// Generate CNI network configuration based on the ENI's desired state.
+	switch eni.DesiredStatus {
+	case status.NetworkReadyPull:
+		cniNetConf = createBranchENIConfig(netNSPath, eni, VPCBranchENIInterfaceTypeVlan)
+	case status.NetworkReady:
+		cniNetConf = createBranchENIConfig(netNSPath, eni, VPCBranchENIInterfaceTypeTap)
+	case status.NetworkDeleted:
+		cniNetConf = createBranchENIConfig(netNSPath, eni, VPCBranchENIInterfaceTypeTap)
+		add = false
+	}
+
+	_, err = c.executeCNIPlugin(ctx, add, cniNetConf)
+	if err != nil {
+		err = errors.Wrap(err, "failed to setup branch eni")
+	}
+
+	return err
+}
+
+// configureAppMesh configures AppMesh in a network namespace.
+// This is used by warmpool and debug-warmpool platforms.
+func (c *common) configureAppMesh(
+	ctx context.Context,
+	netNSPath string,
+	cfg *appmesh.AppMesh,
+) error {
+	c.os.Setenv(CNIPluginLogFileEnv, ecscni.PluginLogPath)
+	defer c.os.Unsetenv(CNIPluginLogFileEnv)
+
+	cniNetConf := createAppMeshPluginConfig(netNSPath, cfg)
+
+	_, err := c.executeCNIPlugin(ctx, true, cniNetConf)
```

Comment:
Q: why appmesh and sc do not have a `DEL`?

Reply:
All appmesh and SC cleanup happens when the netns is deleted. On containerd agent this makes sense because the same host is not reused for subsequent tasks, instead it gets re-imaged.